libs/azure-storage/README.md

```diff
@@ -1,9 +1,184 @@
+)
+
+for doc in loader.lazy_load():
+    print(doc.page_content)
+```
+
+The example below shows how to load documents from a list of blobs in Azure Blob Storage:
+
+```python
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+
+loader = AzureBlobStorageLoader(
+    account_url="http://<my-storage-account-name>.blob.core.windows.net",
+    container_name="<my-container-name>",
+    blob_names=["blob-1", "blob-2", "blob-3"],
+)
+
+for doc in loader.lazy_load():
+    print(doc.page_content)
+```
+
+### Override default credentials
+Below shows how to override the default credentials used by the document loader:
+
+```python
+from azure.core.credentials import AzureSasCredential
+from azure.identity import ManagedIdentityCredential
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+
+# Override with SAS token
+loader = AzureBlobStorageLoader(
+    "https://<my-storage-account-name>.blob.core.windows.net",
+    "<my-container-name>",
+    credential=AzureSasCredential("<sas-token>")
+)
+
+# Override with more specific token credential than the entire
+# default credential chain (e.g., system-assigned managed identity)
+loader = AzureBlobStorageLoader(
+    "https://<my-storage-account-name>.blob.core.windows.net",
+    "<my-container-name>",
+    credential=ManagedIdentityCredential()
+)
+```
+
+### Customizing blob content parsing
+Currently, the default when parsing each blob is to return the content as a single `Document` object with UTF-8 encoding regardless of the file type. For file types that require specific parsing (e.g., PDFs, CSVs, etc.) or when you want to control the document content format, you can provide the `loader_factory` argument to take in an already existing document loader (e.g., PyPDFLoader, CSVLoader, etc.) or a customized loader.
+
+This works by downloading the blob content to a temporary file. The `loader_factory` then gets called with the filepath to use the specified document loader to load/parse the file and return the `Document` object(s).
+
+Below shows how to override the default loader used to parse blobs:
+
+```python
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+from langchain_community.document_loaders import PyPDFLoader
+
+loader = AzureBlobStorageLoader(
+    account_url="http://<my-storage-account-name>.blob.core.windows.net",
+    container_name="<my-container-name>",
+    blob_names="<my-pdf-file.pdf>",
+    loader_factory=PyPDFLoader,
+)
+```
+
+To provide additional configuration, you can define a callable that returns an instantiated document loader as shown below:
+
+```python
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+from langchain_community.document_loaders import PyPDFLoader
+
+def loader_factory(file_path: str) -> PyPDFLoader:
+    return PyPDFLoader(
+        file_path,
+        mode="single",  # Custom configuration
+        pages_delimiter="\n",  # Custom configuration
```

Comment:
Let's just keep the `mode` as `single` and not include the pages delimiter. And maybe for the comment, we can change it to `# To return PDF as a single document instead of extracting documents by page`